app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ActiveRecord::Base
+has_many :tweets
+
+  def slug
+    self.username.downcase.gsub(" ", "-")
+  end
+
+  def authenticate(password)
```

Comment:
if you add the `has_secure_password`macro method to this class that `bcrypt` supplies it includes the authenticate method for you.